src/wmtk/function/simplex/PerSimplexDifferentiableAutodiffFunction.cpp

```diff
@@ -1,63 +1,52 @@
-#include "AutodiffFunction.hpp"
+#include "PerSimplexDifferentiableAutodiffFunction.hpp"
 #include <wmtk/function/utils/AutoDiffRAII.hpp>
+#include <wmtk/function/utils/SimplexGetter.hpp>
 #include <wmtk/simplex/faces_single_dimension.hpp>
 #include <wmtk/simplex/internal/SimplexEqualFunctor.hpp>
+
+
 namespace wmtk::function {
 
-AutodiffFunction::AutodiffFunction(
+PerSimplexDifferentiableAutodiffFunction::PerSimplexDifferentiableAutodiffFunction(
     const Mesh& mesh,
     const PrimitiveType& domain_simplex_type,
     const MeshAttributeHandle<double>& variable_attribute_handle)
     : PerSimplexDifferentiableFunction(mesh, domain_simplex_type, variable_attribute_handle)
 {}
 
-AutodiffFunction::~AutodiffFunction() = default;
+PerSimplexDifferentiableAutodiffFunction::~PerSimplexDifferentiableAutodiffFunction() = default;
 
-auto AutodiffFunction::get_coordinates(
+auto PerSimplexDifferentiableAutodiffFunction::get_coordinates(
     const Tuple& domain_tuple,
     const std::optional<Tuple>& variable_tuple_opt) const -> std::vector<DSVec>
 {
     ConstAccessor<double> pos = mesh().create_const_accessor(get_coordinate_attribute_handle());
     return get_coordinates(pos, domain_tuple, variable_tuple_opt);
 }
-auto AutodiffFunction::get_coordinates(
+auto PerSimplexDifferentiableAutodiffFunction::get_coordinates(
     const ConstAccessor<double>& accessor,
     const Tuple& domain_tuple,
     const std::optional<Tuple>& variable_tuple_opt) const -> std::vector<DSVec>
 {
-    const PrimitiveType primitive_type = get_coordinate_attribute_primitive_type();
-    const std::vector<Tuple> faces = wmtk::simplex::faces_single_dimension_tuples(
+    auto [attrs, index] = utils::get_simplex_vertex_attributes(
```

Comment:
Previously this was over designed to support attributes other than vertices, it would be nice if it remained that way

Reply:
Your implementation of this function didn't actually use the fact that it's a vertex. Just the naming is a bit off

Reply:
yeah it dosent, will it work for any? what name did you suggest?